File: 알고리즘/Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        s = {}
        for ele in nums:
            if(len(s)==0):
                s.add(ele)
                continue
            
            if(ele not in s):
                s.add(ele)
            else:
                return True
        return False

    def containsDuplicate2(self, nums: List[int]) -> bool:
        s = set(nums)
        return True if len(s)!=len(nums) else False

File: 알고리즘/2/Contains Duplicate.cpp
#include <vector>
#include <unordered_set>

using namespace std;

class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {
        unordered_set<int> set;

        for (auto ele : nums)
        {
            if (set.empty())
            {
                set.insert(ele);
                continue;
            }
            auto it = set.find(ele);
            if (it == set.end())
            {
                set.insert(ele);
            }
            else
            {
                return true;
            }
        }

        return false;
    }

    bool containsDuplicate2(vector<int> &nums)
    {
        unordered_set<int> set(nums.begin(), nums.end());

        return (nums.size() != set.size()) ? true : false;
    }
};